lightning/src/chain/mod.rs

```diff
@@ -57,6 +57,81 @@ pub trait Listen {
+///   no longer be in danger of chain reorganization.
+/// - Call [`transaction_unconfirmed`] for any transaction returned by [`get_relevant_txids`] that
+///   has been reorganized out of the chain.
+/// - Call [`best_block_updated`] whenever a new chain tip becomes available.
+///
+/// See individual method documentation for further details.
+///
+/// [`transactions_confirmed`]: Self::transactions_confirmed
+/// [`transaction_unconfirmed`]: Self::transaction_unconfirmed
+/// [`best_block_updated`]: Self::best_block_updated
+/// [`get_relevant_txids`]: Self::get_relevant_txids
+pub trait Confirm {
+	/// Processes transactions confirmed in a block with a given header and height.
+	///
+	/// Should be called for any transactions registered by [`Filter::register_tx`] or any
+	/// transactions spending an output registered by [`Filter::register_output`]. Such transactions
+	/// appearing in the same block do not need to be included in the same call; instead, multiple
+	/// calls with additional transactions may be made.
+	///
+	/// May be called before or after [`best_block_updated`] for the corresponding block. However,
+	/// it must not be called after [`best_block_updated`] if `header` corresponds to a block that
+	/// has been reorganized from the chain (i.e., if the transaction is no longer confirmed).
+	///
+	/// [`best_block_updated`]: Self::best_block_updated
+	fn transactions_confirmed(&self, header: &BlockHeader, txdata: &TransactionData, height: u32);
+
+	/// Processes a transaction that is no longer confirmed as result of a chain reorganization.
+	///
+	/// Should be called for any transaction returned by [`get_relevant_txids`] if it has been
+	/// reorganized out of the best chain. Once called, the given transaction should not be returned
+	/// by [`get_relevant_txids`] unless it has been reconfirmed via [`transactions_confirmed`].
+	///
+	/// [`get_relevant_txids`]: Self::get_relevant_txids
+	/// [`transactions_confirmed`]: Self::transactions_confirmed
+	fn transaction_unconfirmed(&self, txid: &Txid);
+
+	/// Processes an update to the best header connected at the given height.
+	///
+	/// Should be called when a new header is available but may be skipped for intermediary blocks
+	/// if they become available at the same time.
+	///
+	/// Implementations are not required to handle chain reorganizations unless `height` is not
+	/// greater than the height given in the previous call. Instead, reorganizations should be
+	/// handled by [`transaction_unconfirmed`].
+	///
+	/// [`transaction_unconfirmed`]: Self::transaction_unconfirmed
+	fn best_block_updated(&self, header: &BlockHeader, height: u32);
+
+	/// Returns transactions that should be monitored for reorganization out of the chain.
+	///
+	/// Should include any transactions passed to [`transactions_confirmed`] that have insufficient
+	/// confirmations to be safe from a chain reorganization. Should not include any transactions
```

Comment:
Should it precise that "insufficient confirmations" is a matter of user client's policy ? Though `ANTI_REORG_DELAY` isn't configurable for now. A user might be willingly to increase is more funds are at stake or lower it if less funds are at stake.

Lowering increase off-chain liquidity velocity as a node can immediately cancel back HTLC, thus freeing a HTLC slot.

Reply:
Right, not yet configurable. Also, it's general because it's talking about how this may be implemented. For example, `ChannelManager` always returns the funding transactions no matter how many confirmations.